Add release notes for 2021.1.0 Eos. (#118)
vircadia/vircadia-docs-sphinx: docs/source/release-notes/2021-1-0-Eos.md

# Alpha 2021.1.0 Eos Release Notes

This version of Vircadia is a new release. It is **not** compatible with the 2020.3.0 Demeter protocol. This version was released on February 12th, 2021.

## Interface (Codename Athena)

The numbers at the end of each item are the PR numbers in the Vircadia [repo](https://github.com/vircadia/vircadia).

#### General

* UI/UX: The VR controls for regular and fullbody users has been overhauled. (#928)

>     1. Fix being unable to sit in the real world in VR without the avatar and viewpoint being popped up to a standing position (fixes https://github.com/vircadia/vircadia/issues/825). With the new option "Allow my avatar to stand", we can choose to have the avatar stand only when the user is standing. The default behavior is the same as before (always allowed to stand; allowed to lean only when user is standing), except that now, when foot trackers are used, the avatar is never made to stand artificially.

Reviewer: Most of our PRs fix issues so starting to include them seems like a bad idea, unless we do a similar thing like on the PR numbers.
```suggestion
    1. Fix being unable to sit in the real world in VR without the avatar and viewpoint being popped up to a standing position. With the new option "Allow my avatar to stand", we can choose to have the avatar stand only when the user is standing. The default behavior is the same as before (always allowed to stand; allowed to lean only when user is standing), except that now, when foot trackers are used, the avatar is never made to stand artificially.
```

Reviewer 2: Good point. The issue numbers should be in the PRs, anyway.

    2. Fix very wrong walk/fly directions when using foot tracking or when using the mode that allows the avatar to sit when the user sits (previously "Avatar lean behavior: disabled").

>     3. Fix VR walk speeds being too slow at small avatar scale and too fast at large avatar scale (fixes https://github.com/vircadia/vircadia/issues/851). The avatar scale was being factored-in twice in the velocity calculation.

Reviewer: Same as above.
```suggestion
    3. Fix VR walk speeds being too slow at small avatar scale and too fast at large avatar scale. The avatar scale was being factored-in twice in the velocity calculation.
```

    7. Fix for the recentering not behaving as requested by the user after getting up from click-to-sit (always popping up to standing position).
    8. Fix for incorrect vertical position of the avatar and viewpoint after changing recenter mode while not standing in the real world.
    9. The logic controlling the direction of `MyAvatar`'s action motor was hard to follow and maintain, as it was split between two functions. Now it's clearer and all done in `calculateScaledDirection`.
    10. Leaning and lean recentering now work as intended at any avatar scale.
* API: Add `useOriginalPivot` property to model entities allowing entities to use their original pivot. (#951)
* Graphics: Improve rendering of transparent entities. (#953)
* API: Add `billboardMode` property for all entity types allowing entities to face the user as specified in the following modes: none, yaw, full. (#961)
* App: The 'Appreciate' app has been added as a default app to Interface. (#974)
* API: Add `userAgent` property to web entities allowing the web entity to use the specified user agent when browsing. (#998, #1017)

> * App: Allow any user to teleport to any user within a domain using the 'People' app. (#991)

Reviewer: Is this right? Teleport permissions have been removed?

Reviewer 2: Unless I am mistaken, there aren't any teleport permissions per se. Rather there was a limitation in that you had to be logged in order for the "teleport to user" function to work.

Reviewer: No. It is supposed to abide to the permissions set by the user. The setting is still there in v2021.1.0-rc2:
![1Auswahl_846](https://user-images.githubusercontent.com/11144627/107611403-36544380-6c44-11eb-96b0-d97f33a3413e.png)

* API: Add `releaseName` property to the `About` API. (#996)

#### Miscellaneous

* Application: Add functionality to skip repeated log entries. (#937)
* Docs: List required dependency for Visual Studio on Windows builds. (#967)
* Application: Fix multiple crashes when using VR with a QML disabled build. (#971)

> * Application: Vircadia now accepts the `RELEASE_NAME` env variable. (#975)

Reviewer 2: ```suggestion
* Build: Vircadia now accepts the `RELEASE_NAME` env variable. (#975)
```

* Build: Updated Windows VCPKG to include an alternative JOM download URL instead of just Qt's. (#983)
* Build: Revise Windows build docs to improve formatting and list Node.JS as a dependency. (#987)

> * Misc: 'Alpha' references have been removed for Interface indicating that Vircadia is now a full release. (#996)

Reviewer: We are skipping beta? Even though there is tons of bugs and reliability issues?

# Alpha 2021.1.0 Eos Release Notes

> This version of Vircadia is a new release. It is **not** compatible with the 2020.3.0 Demeter protocol. This version was released on February 5th, 2021.

Reviewer 2: ```suggestion
This version of Vircadia is a new release. It was released on February 5th, 2021.

It has a protocol change. It is **not** compatible with the 2020.3.n Demeter protocol. 
```
Also, update date at time of release.

    10. Leaning and lean recentering now work as intended at any avatar scale.
* API: Add `useOriginalPivot` property to model entities allowing entities to use their original pivot. (#951)
* Graphics: Improve rendering of transparent entities. (#953)
* API: Add `billboardMode` property for all entity types allowing entities to face the user as specified in the following modes: none, yaw, full. (#961)
* App: The 'Appreciate' app has been added as a default app to Interface. (#974)
* API: Add `userAgent` property to web entities allowing the web entity to use the specified user agent when browsing. (#998)
* App: Allow any user to teleport to any user within a domain using the 'People' app. (#991)
* API: Add `releaseName` property to the `About` API. (#996)

#### Miscellaneous

* Application: Add functionality to skip repeated log entries. (#937)
* Docs: List required dependency for Visual Studio on Windows builds. (#967)
* Application: Fix multiple crashes when using VR with a QML disabled build. (#971)
* Application: Vircadia now accepts the `RELEASE_NAME` env variable. (#975)

> * Build: Updated Windows VCPKG to use an alternative JOM download URL instead of Qt's. (#983)

Reviewer 2: ```suggestion
* Build: Updated Windows VCPKG to include an alternative JOM download URL instead of just Qt's. (#983)
```

The numbers at the end of each item are the PR numbers in the Vircadia [repo](https://github.com/vircadia/vircadia).

#### General

* Nothing new to report!

## Metaverse Server & Dashboard (Codename Iamus)

### Iamus [Metaverse Server](https://github.com/vircadia/iamus)

### Iamus [Metaverse Dashboard](https://github.com/vircadia/project-iamus-dashboard)

Updates are reflected [here](https://dashboard.vircadia.com/). The Dashboard has been upgraded from version 1.2.2 to version 1.2.4.

* Fix token refresh routing & logout bug.
* Fix general navigation issues.
* Fix profile information loading.
* Add API 'Request' stats component to the home page.
* Fix a page loading style bug.
* The menu's navigation now highlights based on what page you are on.

## Vircadia Launcher (Codename Pantheon)

[Vircadia's launcher](https://github.com/vircadia/pantheon-launcher) updates can be found [here](https://github.com/vircadia/pantheon-launcher/commits/master). The launcher is on version 1.0.1.

* Enable tray functionality on launcher.
* Set 'News' tab as default.
* Add 'Metaverse' tab.
* Add 'Hide On Launch' option to minimize the launcher to tray when Interface is started successfully.

> * Various backend & stability fixes have been implemented.

Reviewer 2: ```suggestion
* Various backend & stability fixes.
```

    10. Leaning and lean recentering now work as intended at any avatar scale.
* API: Add `useOriginalPivot` property to model entities allowing entities to use their original pivot. (#951, #1029)
* Graphics: Improve rendering of transparent entities. (#953)
* API: Add `billboardMode` property for all entity types allowing entities to face the user as specified in the following modes: none, yaw, full. (#961, #1041)
* App: The 'Appreciate' app has been added as a default app to Interface. (#974)
* API: Add `userAgent` property to web entities allowing the web entity to use the specified user agent when browsing. (#998, #1017)
* App: Allow any user to teleport to any user within a domain using the 'People' app. (#991)
* API: Add `releaseName` property to the `About` API. (#996)
* API: Add `alignment` property to text entities allowing text to be aligned to 'left', 'center', and 'right'. (#1020)
* App: 'Create' app February 2021 improvements package. (#1030)
    * The Entity List menubar has been reorganized.

>         - Introducing a new menu “**Transform**”. The menu items have been re-dispatched. CSS has been retouched to display clearly which menu is open.

Reviewer 2: ```suggestion
        - Introducing a new menu “Transform”. The menu items have been re-dispatched. CSS has been retouched to display clearly which menu is open.
```
